dags/synapse-dataset-to-croissant.py

```diff
@@ -165,17 +199,552 @@ def set_up_logging() -> Logger:
+    database in Snowflake.
+
+    Arguments:
+        files_attached_to_dataset: The list of files attached to the dataset.
+        context: The context of the DAG run.
+
+    Returns:
+        The distribution section for the files attached to the dataset.
+    """
+    distribution_files = []
+    files_to_find_md5_in_snowflake = {}
+
+    ids_of_files = []
+    id_and_version_pairs = []
+    for file in files_attached_to_dataset:
+        file: File = file
+        if not file.file_handle or not file.file_handle.content_md5:
+            modified_file_id = int(file.id.replace("syn", ""))
+            ids_of_files.append(modified_file_id)
+            id_and_version_pairs.append(modified_file_id)
+            id_and_version_pairs.append(file.version_number)
+            files_to_find_md5_in_snowflake[int(file.id.replace(
+                "syn", ""))] = file.version_number
+
+    file_md5s = None
+    if ids_of_files:
+        snow_hook = SnowflakeHook(
+            context["params"]["snowflake_developer_service_conn"])
+        ctx = snow_hook.get_conn()
+        cs = ctx.cursor()
+
+        query = f"""
+        WITH version_data AS (
+            SELECT
+                nl.id,
+                flattened.value:versionNumber::int AS versionNumber,
+                flattened.value:contentMd5::string AS contentMd5
+            FROM synapse_data_warehouse.synapse.node_latest AS nl,
+            LATERAL FLATTEN(input => nl.VERSION_HISTORY) AS flattened
+            WHERE nl.id IN ({', '.join(['%s'] * len(ids_of_files))})
+            AND nl.is_public = TRUE
+        )
+        SELECT
+            vd.id,
+            vd.versionNumber,
+            vd.contentMd5
+        FROM version_data AS vd
+        WHERE (vd.id, vd.versionNumber) IN ({', '.join(['(%s, %s)'] * (len(id_and_version_pairs)//2))});
+        """
+        try:
+            cs.execute(
+                query,
+                (ids_of_files + id_and_version_pairs),
+            )
+            file_md5s = cs.fetch_pandas_all()
+        finally:
+            cs.close()
+
+    for file in files_attached_to_dataset:
+        file: File = file
+        if file.file_handle and file.file_handle.content_md5:
+            file_md5 = file.file_handle.content_md5
+        elif not file_md5s.empty and int(file.id.replace("syn", "")) in file_md5s["ID"].values:
+            file_md5 = file_md5s.loc[file_md5s["ID"] == int(
+                file.id.replace("syn", "")), "CONTENTMD5"].values[0]
+        else:
+            file_md5 = "unknown_md5"
+
+        distribution_files.append(
+            {
+                "@type": "FileObject",
+                "@id": f"{file.id}.{file.version_number}",
+                "name": f"{file.name}",
+                "description": f"Data file associated with {file.name}",
+                "contentUrl": f"https://www.synapse.org/Synapse:{file.id}.{file.version_number}",
+                "encodingFormat": "application/json",
```

Comment:
Is this encoding format for the associated file...?

Reply:
Good point. I added it to the ticket and will look to pull the right format based on the file.

Reply:
Done in https://github.com/Sage-Bionetworks-Workflows/orca-recipes/pull/84